### PR TITLE
29 create transaction history component

### LINKED ISSUE #29: Create transaction history component
**Description:**

- Implement transaction history component on home page.
- Show three types of transactions: sent, received, and reward.
- /wallet/history/<walletId> GET but with walletId passed as query parameter.
- payload from service: 

```json
{
   [
      {
         "transaction": Transaction,
         "type": "send" or "receive" or "reward"
      },
      {
             ...
      }
   ]
}
```

**Acceptance Criteria:**

- [x] Transaction history component
- [x] Hit service for new transaction history when the user sends or receives a transaction

### PR DESCRIPTION
**Pull request by madisonfines**
Implemented transaction history component on the home page. Click the component to see transaction history.

closes #29

### CODE REVIEW

File: src/frontend/src/store/index.ts [hunk 1]
```diff
@@ -404,5 +411,53 @@ export default new Vuex.Store({
...
+          },
+          (err) => {
+            const message = err.response && err.response.data.error
+              ? err.response.data.error
+              : ERROR_STRING
+            dispatch("ACTION_DISPLAY_TOAST", { message: message, type: 'error' })
+          }
+        ); 
+ 
+      /* const transactions = [
```
Comment (madisonfines) [inline src/frontend/src/store/index.ts]: To be removed, used for testing

File: src/frontend/src/store/index.ts [hunk 2]
```diff
@@ -404,5 +411,53 @@ export default new Vuex.Store({
         dismissible: true,
       });
     },
+
+    ACTION_FETCH_TRANSACTION_HISTORY({ commit, getters, dispatch }) {
+      axios
+        .get("http://localhost//wallet/history?walletId="+ getters.walletId)
+        .then(
+          (response) => {
+            commit("MUTATION_SET_TRANSACTION_HISTORY", response.data);
```
Comment (ZackHolmberg) [inline src/frontend/src/store/index.ts]: Verify this will work with backend implementation when it is up

**Review by ZackHolmberg — APPROVED** [review]
nice

File: src/frontend/src/store/index.ts [hunk 3]
```diff
@@ -404,5 +411,53 @@ export default new Vuex.Store({
         dismissible: true,
       });
     },
+
+    ACTION_FETCH_TRANSACTION_HISTORY({ commit, getters, dispatch }) {
+      axios
+        .get("http://localhost//wallet/history?walletId="+ getters.walletId)
```
Comment (akshaysharma21) [inline src/frontend/src/store/index.ts]: Gonna need to update this to the backend route without the payload

**Review by akshaysharma21 — CHANGES_REQUESTED** [review]

File: src/frontend/src/store/index.ts (hunk 3 above)
Comment (abhisachdev17) [inline src/frontend/src/store/index.ts]: you will need to change this to `http://localhost/wallet/history/ + getters.walletId`

**Review by abhisachdev17 — APPROVED** [review]
Just one change, rest looks good

File: src/frontend/src/store/index.ts (hunk 3 above)
Comment (abhisachdev17) [reply src/frontend/src/store/index.ts]: This reminded me the walletId doesn't exist anymore. You will need to use getters.umnetId

File: src/frontend/src/store/index.ts [hunk 4]
```diff
@@ -366,5 +373,21 @@ export default new Vuex.Store({
         dismissible: true,
       });
     },
+
+    ACTION_FETCH_TRANSACTION_HISTORY({ commit, getters, dispatch }) {
+      axios
+        .get("http://localhost//wallet/history/"+ getters.umnetId)
```
Comment (ZackHolmberg) [inline src/frontend/src/store/index.ts]: Extra `/` here in `localhost//wallet`, not sure if it actually affects the success of the call or not

**Review by akshaysharma21 — APPROVED** [review]

File: src/frontend/src/components/TransactionHistory.vue [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    return this.$store.getters.transactions;
+  }
+
+  collapse() {
+    if(this.$data.active) {
+      this.$data.active = false;
+    }
+    else {
+      this.$data.active = true;
+    }
```
Comment (ZackHolmberg) [inline src/frontend/src/components/TransactionHistory.vue]: I think this can just be one-lined as `this.$data.active = !this.$data.active`

**Review by ZackHolmberg — APPROVED** [review]

**Pull request merged**